### PR TITLE
Pages docs

### PR DESCRIPTION
**Pull request by jacobtoppm**
Adds a basic mkdocs site, extending and replacing the readme documentation with
- More recipes (for creating a new Picker, and custom converters)
- Info on Microsoft setup
- General rewording

### CODE REVIEW

File: docs/basic_usage.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
+# Basic Usage
+
+1. To enable import for a Page model, it should inherit from `ContentImportMixin` 
+(`wagtail_content_import.models.ContentImportMixin`). By default, content will be imported into into a StreamField called body
```
Comment (tomdyson) [inline docs/basic_usage.md]: body -> 'body'

File: docs/basic_usage.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+            table = TableConverter('my_table_block')    
+   
+      This would map to an example StreamField defined as:
+   
+        from wagtail.images.blocks import ImageChooserBlock
+        from wagtail.core.blocks import CharBlock, RichTextBlock, StreamBlock
+        from wagtail.contrib.table_block.blocks import TableBlock
+        
+        class BaseStreamBlock(StreamBlock):
+        """
```
Comment (tomdyson) [inline docs/basic_usage.md]: doc strings should be indented

File: docs/getting_started.md [hunk 3]
```diff
@@ -0,0 +1,24 @@
+## Requirements:
+* Django 2.2
+* Wagtail 2.2
+* Python 3
```
Comment (tomdyson) [inline docs/getting_started.md]: Python 3 is required by Wagtail 2+ anyway

File: docs/getting_started.md [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+
+### To set up:
+ 1. Run `python3 pip install wagtail-content-import`.
+ 2. Add `'wagtail_content_import'` to `INSTALLED_APPS` above `wagtail.admin`
+ 3. Add the urls: include `wagtail_import.urls` in your `urlpatterns` in `urls.py`. This could look like:
+    
+        from django.conf.urls import include, url
```
Comment (tomdyson) [inline docs/getting_started.md]: Format as ```python ?

File: docs/index.md [hunk 5]
```diff
@@ -0,0 +1,19 @@
+# Welcome to Wagtail Content Import's Documentation
+
+Wagtail Content Import is a module for Wagtail that provides functionality for importing page content from 
+third-party sources. Page content is imported into a StreamField, using a set of customisable mappings.
```
Comment (tomdyson) [inline docs/index.md]: "for Wagtail that provides functionality for importing page content from third-party sources" -> "for importing page content into Wagtail from third-party sources"

File: docs/submitting_backend.md [hunk 6]
```diff
@@ -0,0 +1,205 @@
+Thanks for being interested in contributing! All contributions should be submitted as pull requests
+to the [Wagtail Content Import repository](https://github.com/torchbox/wagtail-content-import).
+
+
+## Submitting a New Picker
+
+If you're planning on submitting a new picker - apps which enable choosing and importing a file from a 
+remote source - you'll need to follow this blueprint:
+
+###Overview and File Structure
```
Comment (tomdyson) [inline docs/submitting_backend.md]: I think you need a space between ### and title (here and for subtitles below)

File: docs/getting_started.md (hunk 4 above)
Comment (jacobtoppm) [reply docs/getting_started.md]: Unfortunately, this seems to be one of the quirks of mkdocs (https://www.mkdocs.org/user-guide/writing-your-docs/#writing-with-markdown, under "Fenced code blocks"): ```python blocks can't be indented, so I had to choose list formatting or syntax highlighting.

I thought it was useful having it as part of the numbered instructions, but happy to move it elsewhere and get the highlighting back if you think that would be better.

File: docs/basic_usage.md (hunk 1 above)
Comment (jacobtoppm) [reply docs/basic_usage.md]: Thanks, fixed in latest commit!

File: docs/basic_usage.md (hunk 2 above)
Comment (jacobtoppm) [reply docs/basic_usage.md]: Thanks - have indented now.

File: docs/getting_started.md (hunk 3 above)
Comment (jacobtoppm) [reply docs/getting_started.md]: Good point, I've removed now.

File: docs/index.md (hunk 5 above)
Comment (jacobtoppm) [reply docs/index.md]: Yes, that sounds better, thanks - I've updated in the latest version.

File: docs/submitting_backend.md (hunk 6 above)
Comment (jacobtoppm) [reply docs/submitting_backend.md]: Added now, thanks

File: docs/getting_started.md (hunk 4 above)
Comment (tomdyson) [reply docs/getting_started.md]: Ah right, thanks for the explanation! Let's leave it like this.

**Pull request merged**